php/shared/usr/local/share/php/common_functions.sh

```diff
@@ -60,3 +60,33 @@ composer_package_compare() {
   dpkg --compare-versions "${PACKAGE_VERSION}" "${RELATION}" "${COMPARE_VERSION}"
   return "$?"
 }
+
+parse_uri() {
+  # extract the protocol
+  proto="`echo $1 | grep '://' | sed -e's,^\(.*://\).*,\1,g'`"
+  scheme="`echo $proto | cut -d':' -f1`"
+
+  # remove the protocol
+  url=`echo $1 | sed -e s,$proto,,g`
+
+  # extract the user and password (if any)
+  userpass="`echo $url | grep @ | cut -d@ -f1`"
+  pass=`echo $userpass | grep : | cut -d: -f2`
+  if [ -n "$pass" ]; then
+    user=`echo $userpass | grep : | cut -d: -f1`
+  else
+    user=$userpass
+  fi
+
+  # extract the host -- updated
+  hostport=`echo $url | sed -e s,$userpass@,,g | cut -d/ -f1`
+  port=`echo $hostport | grep : | cut -d: -f2`
+  if [ -n "$port" ]; then
+    host=`echo $hostport | grep : | cut -d: -f1`
+  else
+    host=$hostport
+  fi
+
+  # extract the path (if any)
+  path="`echo $url | grep / | cut -d/ -f2-`"
```

Comment:
I'd still like to see the php parseurl used, just as it's one less thing that could be wrong.

Also Kieran is thinking about using some concurency, which could cause issues with these global proto/scheme/user/password/path will have issues with unless passed as a sub-shell

Reply:
not that I know a good way to use parseurl and collect the arguments though

Reply:
OK, that's fair enough. Then... how would the result of this function look like then? /cc @kierenevans